atc/resource/resource_instance_fetch_source.go

```diff
@@ -139,6 +138,7 @@ func (s *resourceInstanceFetchSource) Create(ctx context.Context) (VersionedSour
 		s.session.Metadata,
 		s.containerSpec,
 		s.resourceTypes,
+		nil,
```

Comment:
We should be passing the lock in here since we could be creating a `db.CreatingContainer` inside `FindOrCreateContainer`.